seismicpro/src/weathering_velocity.py

```diff
@@ -0,0 +1,262 @@
+
+    Parameters
+    ----------
+    offsets : ndarray
+        offsets of a traces
+    max_offset : int
+        maximum offsets value
+    picking_times : ndarray
+        picking times of a traces
+    init : dict
+        inital values for fitting a piecewise function. Used to calculate `bounds` and `n_layers` if these params not enough.
+    bounds : Dict[List]
+        left and right bounds for any parameter of a piecewise function. Used to calculate `init` and `n_layers` if these params not enough.
+    n_layers : int
+        prior quantity of a layers of a weathering model. Interpreting like a quantity piece of a piecewise funtion.
+        Used for calculate `init` and `bounds` if these params not enough.
+
+    Attributes
+    ----------
+    offsets : ndarray
+        contains offsets of a traces.
+    max_offset : int
+        maximum offsets value.
+    picking_times : ndarray
+        picking times of a traces.
+    init : dict
+        inital values used for fitting a piecewise function.
+    bounds : Dict[List]
+        left and right bounds used for fitting a piecewise function.
+    n_layers : int
+        quantity piece of a fitted piecewise funtion.
+        # check naming with Dan
+    params : dict
+        contains fitted values of a piecewise function.
+    _n_iters : int
+        quantity of iteration for fitting a piecewise function.
+
+    Raises
+    ------
+    ValueError
+        if any `init` values is negative
+        if any `bounds` values is negative
+        if left bound greater than right bound
+        if passed `init` and/or `bounds` keys are insufficient or excess
+
+    """
+
+    def __init__(self, offsets, picking_times, n_layers=None, init=None, bounds=None, **kwargs):
+
+        init = {} if init is None else init
+        bounds = {} if bounds is None else bounds
+
+        self.offsets = offsets
+        self.max_offset = offsets.max()
+        self.picking_times = picking_times
+
+        self._check_values(init, bounds, n_layers)
+
+        self.init = {**self._calc_init_by_layers(n_layers), **self._calc_init_by_bounds(bounds), **init}
+        self.bounds = {**self._calc_bounds_by_init(self.init), **bounds}
+        self.n_layers = len(self.bounds) // 2
+
+        self._check_keys()
+
+        # piecewise func variables
+        self._n_iters = 0
+        self._piecewise_times = np.empty(self.n_layers + 1)
+        self._piecewise_offsets = np.zeros(self.n_layers + 1)
+        self._piecewise_offsets[-1] = self.max_offset
+
+        # Piecewise linear regression minimization
+        constraints = {"type": "ineq", "fun": lambda x: (np.diff(x[1:self.n_layers]) >= 0).all(out=np.array(0))}
+        minimizer_kwargs = {'method': 'SLSQP', 'constraints': constraints, **kwargs}
+        model_params = optimize.minimize(self.piecewise_linear, x0=self._stack_values(self.init),
+                                         bounds=self._stack_values(self.bounds), **minimizer_kwargs)
```

Comment:
List for x0 or list of lists for bounds are enough for scipy. 
As far as I understand `_stack_values` are more about ordering values. Just secure the proper order of `self.init` and `self.bounds`  in `self.check_keys`

```suggestion
        model_params = optimize.minimize(self.piecewise_linear, x0=list(self.init.values()),
                                         bounds=list(self.bounds.values()), **minimizer_kwargs)
```

Reply:
add reodrdering dicts and `_stack_values()` are removed